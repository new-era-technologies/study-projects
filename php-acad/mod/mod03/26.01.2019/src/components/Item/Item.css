.Item__title {
    font-size: 18px;
    font-weight: bold;
}

.Item__tag {
    padding-left: 5px;
    font-size: 12px;
    font-weight: bold;
}