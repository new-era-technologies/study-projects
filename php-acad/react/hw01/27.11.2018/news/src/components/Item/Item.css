.Item {
     width: 30%;
 }
.Item__link {
    text-decoration: none;
}
.Item__link__pic {
    width: 100%;
    height: 260px;
}
.Item__link__title {
    height: 40px;
    overflow: hidden;
    color: black;
    text-align: left;
}
.Item__link__descr {
    height: 65px;
    overflow: hidden;
    color: black;
    text-align: left;
}
.Item__link__sign {
    display: flex;
    justify-content: space-between;
}
.Item__link__sign__date {
    color: black;
}
.Item__link__sign__author {
    color: #1b5288;
}