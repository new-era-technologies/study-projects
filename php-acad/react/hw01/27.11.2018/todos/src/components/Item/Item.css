.Item {
    display: flex;
    justify-content: start;
    align-items: center;
    border-bottom: 1px solid #F7F7F7;
    padding: 10px 0;
}
.Item__btn {
    background-color: white;
    color: #E7D8DD;
    border: 2px solid #E7D8DD;
    border-radius: 4px;
    box-shadow: 0 0 1px 0 #E7D8DD;
    outline: none;
    padding: 5px 10px;
}
.Item__text {
    margin: 0 0 0 15px;
    font-weight: 500;
}