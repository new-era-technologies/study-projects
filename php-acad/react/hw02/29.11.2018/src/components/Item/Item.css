.Item {
    width: 30%;
}
.Item__link {
    text-decoration: none;
}
.Item__link__title {
    height: 40px;
    overflow: hidden;
    color: black;
    text-align: left;
}
.Item__link__pic {
    width: 100%;
}
.Item__link__descr {
    height: 45px;
    overflow: hidden;
    color: black;
    text-align: left;
}
.Item__link__date {
    margin: 0 0 10px 0;
    color: black;
    text-align: right;
}