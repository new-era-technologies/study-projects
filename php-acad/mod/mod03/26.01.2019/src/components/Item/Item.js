import React, {Component} from 'react';
import './Item.css';


/**
 * Class representing an item.
 * @extends Component
 */
class Item extends Component {
    render() {
        /**
         * @memberof containers.App
         * @prop {Object} the props that are passed to this component
         * @prop {string} title - the name of my component
         * @prop {array} tag - the array of tags of my component
         * @prop {string} tag - the names of tags in array
         */
        const {
            props: {
                item: {
                    title,
                    tag
                }
            }
        } = this;
        /**
         * @description Component Item with title and list of tags
         * @return {string} - HTML markup for the component
         */
        return (
            <div className='Item'>
                <p className='Item__title'>{title}</p>
                {
                    /**
                     * @function the map() method
                     * @description a function to be run for each element in array
                     * @param {string} the value of the current element
                     * @param {number} the array index of the current element
                     * @return {string} HTML markup
                     */
                    tag.map((item, i) => (
                    <span className='Item__tag' key={i}>{item}</span>
                ))}
            </div>
        )
    }
}

export default Item;