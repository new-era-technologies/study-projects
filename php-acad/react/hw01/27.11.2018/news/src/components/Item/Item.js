import React, { Component } from 'react';
import './Item.css';

class Item extends Component {
    render() {
        const {
            props: {
                item: {
                    url,
                    urlToImage,
                    title,
                    description,
                    publishedAt,
                    author
                }
            }
        } = this;
        return (
            <div className="Item">
                <a href={url} className="Item__link" target='_blank' rel="noopener noreferrer">
                    <img src={urlToImage} alt={title} className="Item__link__pic"/>
                    <h4 className="Item__link__title">{title}</h4>
                    <p className="Item__link__descr">{description}</p>
                    <p className="Item__link__sign">
                        <span className="Item__link__sign__date">{publishedAt}</span>
                        <span className="Item__link__sign__author">{author}</span>
                    </p>
                </a>
            </div>
        );
    }
}

export default Item;