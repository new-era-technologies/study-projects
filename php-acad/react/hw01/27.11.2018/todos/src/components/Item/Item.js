import React, {Component} from 'react';
import './Item.css';

class Item extends Component {
    render() {
        const {
            props: {
                item: {id, text},
                deleteItem
            }
        } = this;
        return (
            <div>
                <div className='Item'>
                    <button className='Item__btn' onClick={() => deleteItem(id)}>remove</button>
                    <p className='Item__text'>{text}</p>
                </div>
            </div>
        )
    }
}

export default Item;