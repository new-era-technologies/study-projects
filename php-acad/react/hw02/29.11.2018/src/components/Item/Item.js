import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import './Item.css';

class Item extends Component {
    render() {
        const {
            props: {
                item: {
                    id,
                    title,
                    short_description,
                    social_image,
                    past_time
                }
            }
        } = this;
        return (
            <div className="Item">
                <Link to={'/Item' + id} className='Item__link'>
                    <h4 className="Item__link__title">{title}</h4>
                    <img src={social_image} alt={title} className="Item__link__pic"/>
                    <p className="Item__link__descr">{short_description}</p>
                    <p className="Item__link__date">{past_time}</p>
                </Link>
            </div>
        );
    }
}

export default Item;