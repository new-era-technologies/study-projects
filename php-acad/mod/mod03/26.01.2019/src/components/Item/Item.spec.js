import React from 'react';
import {shallow} from 'enzyme';
import Item from './Item';


describe('Item Container', () => {

    const props = {
        item: {
            title: 'SDK',
            tag: [
                'Tag1',
                'Tag2'
            ]
        }
    };

    const ItemContainer = shallow(<Item {...props} />);

    it('renders properly', () => {
        expect(ItemContainer).toMatchSnapshot()
    });

    describe('Correct render for first SDK Item', () => {

        it('title', () => {
            expect(
                ItemContainer
                    .find('p')
                    .text()
            ).toEqual('SDK')
        });

        it('tag', () => {
            expect(
                ItemContainer
                    .find('span')
                    .first()
                    .text()
            ).toEqual('Tag1')
        })
    })

});