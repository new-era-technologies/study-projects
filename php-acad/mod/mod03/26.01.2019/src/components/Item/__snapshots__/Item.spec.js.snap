// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Container renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    item={
      Object {
        "tag": Array [
          "Tag1",
          "Tag2",
        ],
        "title": "SDK",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="Item__title"
        >
          SDK
        </p>,
        Array [
          <span
            className="Item__tag"
          >
            Tag1
          </span>,
          <span
            className="Item__tag"
          >
            Tag2
          </span>,
        ],
      ],
      "className": "Item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SDK",
          "className": "Item__title",
        },
        "ref": null,
        "rendered": "SDK",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Tag1",
          "className": "Item__tag",
        },
        "ref": null,
        "rendered": "Tag1",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Tag2",
          "className": "Item__tag",
        },
        "ref": null,
        "rendered": "Tag2",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="Item__title"
          >
            SDK
          </p>,
          Array [
            <span
              className="Item__tag"
            >
              Tag1
            </span>,
            <span
              className="Item__tag"
            >
              Tag2
            </span>,
          ],
        ],
        "className": "Item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SDK",
            "className": "Item__title",
          },
          "ref": null,
          "rendered": "SDK",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Tag1",
            "className": "Item__tag",
          },
          "ref": null,
          "rendered": "Tag1",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Tag2",
            "className": "Item__tag",
          },
          "ref": null,
          "rendered": "Tag2",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
